x/msgfee/codec.proto

```diff
@@ -0,0 +1,13 @@
+syntax = "proto3";
+
+package msgfee;
+
+import "github.com/iov-one/weave/coin/codec.proto";
+import "github.com/gogo/protobuf/gogoproto/gogo.proto";
+
+// MsgFee represents a fee for a single message that must be paid in order for
+// the message to be processed.
+message MsgFee {
+	string msg_path = 1;
+	coin.Coin fee = 2;
```

Comment:
Q: Would it make sense to have the not nil annotation for fee?

Reply:
pp: `fee` is a bit overloaded already. This is the product fee to be more precise

Reply:
Yes, it does make sense to change it to a non-pointer type. :running_man: 

Now that you pointed it out I agree.  I think `ProductFee` is not describing it well. The functionality is to charge for a transaction processing, there is no product :thinking:
 
How about simple `Value`? `MsgFee.Value` of type `coin.Coin`.

Reply:
I like to think of our features as products especially if they come with a `price`. I can not think of a better name now. Sorry. If you don't like price then please use `fee`. It is better than `value` as it indicates somebody has to pay for this, too. But don't feel blocked by this.

Reply:
I will stick to the `Fee` then. 
I can be very wrong because I am not a native speaker. `price` sounds to me like I am paying coins for a product. `fee` sounds like I am paying coins in addition to the `price` - an extra amount required as an addition to the transaction.

In real world when I buy a house the price is 1mln EUR plus fees - sales person, writing to the mortgage register, tax. The 1mln is the `price` and all others are `fee`. :thinking: